Formula/rat.rb

```diff
@@ -0,0 +1,26 @@
+class Rat < Formula
+  desc "Compose shell commands to build terminal applications"
+  homepage "https://github.com/ericfreese/rat"
+  url "https://github.com/ericfreese/rat.git"
+  version "0.0.2"
+  sha256 "9da8d67a1e16056e3f05cdf536766e5571ede54a6c241c941d6c74eee5d64d31"
+
+  depends_on "go" => :build
+
+  def install
+    contents = Dir["{*,.git,.gitignore}"]
+    gopath = buildpath/"gopath"
+    (gopath/"src/github.com/ericfreese/rat").install contents
+
+    ENV["GOPATH"] = gopath
+
+    cd gopath/"src/github.com/ericfreese/rat" do
+      system "go", "build", "-o", "rat"
+      bin.install "rat"
+    end
+  end
+
+  test do
+    system "[", "-x", "#{bin}/rat", "]"
```

Comment:
What is this doing?

Reply:
Checking that the file exists and is executable.

Reply:
Can an actual functionality test be done?

Reply:
I don't think so. Running the program connects to the terminal and expects user input. I suppose I could run `rat --help` and test that it succeeds. What that count as a "functionality test"?

Reply:
Can we fake user input using `expect` [like this](https://github.com/Homebrew/homebrew-core/blob/7968383b22366ab73b584fb3370778d09d005e47/Formula/minisign.rb#L26-L39), or?

Reply:
Possibly. It uses termbox, an ncurses-like library, for rendering to the terminal so it depends how sophisticated `expect` is in terms of terminal screen-scraping. I see [`tig`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/tig.rb) is just executing a version check. `rat` doesn't currently support a `--version` flag.

Reply:
> rat doesn't currently support a --version flag.

It might be worth asking for one. Homebrew is pretty strict on functionality testing these days but at the very least anything is better than simply checking the file exists & is executable.

Reply:
Ok. I actually was digging into `expect` yesterday and I think I should be able to put something together using it.